containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -932,13 +970,25 @@ class UnorderedMap {
   friend struct Impl::UnorderedMapPrint;
 };
 
-// Specialization of deep_copy for two UnorderedMap objects.
+// Specialization of deep_copy() for two UnorderedMap objects.
 template <typename DKey, typename DT, typename DDevice, typename SKey,
           typename ST, typename SDevice, typename Hasher, typename EqualTo>
 inline void deep_copy(
     UnorderedMap<DKey, DT, DDevice, Hasher, EqualTo> &dst,
     const UnorderedMap<SKey, ST, SDevice, Hasher, EqualTo> &src) {
-  dst.create_copy_view(src);
+  dst.deep_copy_view(src);
```

Comment:
Assuming `KOKKOS_ENABLE_DEPRECATED_CODE_4` is enabled, this is actually a change in behavior and will not work as expected if using the `deep_copy` to also perform the view allocations on the destination `UnorderedMap`.

Reply:
Scratch that, I see the `allocate_view` call here: https://github.com/kokkos/kokkos/pull/6812/files#diff-2c8acfc1ac5cbbb4cc3da14e466f42c85e415b540c815bf2630c42ec2ac09d59R864.